python/ambassador/config/resourcefetcher.py

```diff
@@ -261,63 +271,59 @@ def check_k8s_dup(self, kind: str, namespace: str, name: str) -> bool:
         self.k8s_parsed[key] = True
         return True
 
-    def handle_k8s(self, obj: dict) -> None:
+    def handle_k8s(self, raw_obj: dict) -> None:
         # self.logger.debug("handle_k8s obj %s" % json.dumps(obj, indent=4, sort_keys=True))
 
-        kind = obj.get('kind')
-
-        if not kind:
-            # self.logger.debug("%s: ignoring K8s object, no kind" % self.location)
+        try:
+            obj = KubernetesObject(raw_obj, default_namespace='default')
+        except ValueError:
+            # The object doesn't contain a kind, API version, or name, so we
+            # can't process it.
             return
 
-        metadata = obj.get('metadata') or {}
-        name = metadata.get('name') or '(no name?)'
-        namespace = metadata.get('namespace') or 'default'
+        with self.manager.locations.push_reset():
+            if self.k8s_processor.try_process(obj):
+                # Nothing else to do.
+                return
```

Comment:
My plan is to fast-follow this PR with a few more PRs that move Kubernetes object handling from the fetcher to individual KubernetesProcessor subclasses. So this will eventually be the end of this method.